internal/buffer/search.go

```diff
@@ -239,5 +238,6 @@ func (b *Buffer) ReplaceRegex(start, end Loc, search *regexp.Regexp, replace []b
 
 	b.MultipleReplace(deltas)
 
-	return found, charCount - charsEnd
+	deltaX := util.CharacterCount(b.LineBytes(end.Y)) - charCount
```

Comment:
As I already noted, if the loop was never executed (so nothing was replaced), we should return 0, but here we return non-zero?

Also please squash such bugfixes into previous commits. The commit history in a PR should be like "fix issue A, fix issue B", not like "fix issue A, fix issue B, introduce issue C, fix issue C".

Reply:
I did a separate commit so that you can see the differences more easily. I'm going to squash it.

I think there is always at least one iteration of the loop because we make sure that `start` is less than or equal to `end`. Do you want me to change the code anyway, maybe because you feel it's safer otherwise?

Reply:
> I did a separate commit so that you can see the differences more easily. I'm going to squash it.

Github allows to see the diff for force-pushes.

> I think there is always at least one iteration of the loop because we make sure that `start` is less than or equal to `end`.

Ah sorry, missed that.

> Do you want me to change the code anyway, maybe because you feel it's safer that way?

I think yes, it's better to make it explicit (i.e. `if i == end.Y` in the last iteration, like it was before), it makes the code cleaner and easier to understand.

BTW what's wrong with the original name `netrunes`?

Reply:
> what's wrong with the original name `netrunes`?

We are not counting runes, but characters that are displayed. [Combining characters](https://en.wikipedia.org/wiki/Combining_character) encoded in additional runes are not counted.

Reply:
Yes, but that's a rather pedantic distinction. I think we are using "rune" in both meanings interchangeably in various places.

Anyway, nevermind.